docs/api/locking.md

```diff
@@ -0,0 +1,449 @@
+# Git LFS File Locking API
+
+Added: v2.0
+
+The File Locking API is used to create, list, and delete locks, as well as
+verify that locks are respected in Git pushes. The locking URLs are built
+by adding a suffix to the LFS Server URL.
+
+Git remote: https://git-server.com/foo/bar  
+LFS server: https://git-server.com/foo/bar.git/info/lfs  
+Locks API: https://git-server.com/foo/bar.git/info/lfs/locks
+
+See the [Server Discovery doc](./server-discovery.md) for more info on how LFS
+builds the LFS server URL.
+
+All File Locking requests require the following HTTP headers:
+
+    Accept: application/vnd.git-lfs+json
+    Content-Type: application/vnd.git-lfs+json
+
+See the [Authentication doc](./authentication.md) for more info on how LFS
+gets authorizes Batch API requests.
+
+Note: This is the first version of the File Locking API, supporting only the
```

Comment:
Perhaps we should `>` quote this section since it's a `Note:` and temporary. Would also be nice to distinguish this section from the rest of the documentation.